src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -301,8 +303,12 @@ internal void ToString(TraceFormat traceFormat, StringBuilder sb)
                             if (pi[j].ParameterType != null)
                                 typeName = pi[j].ParameterType.Name;
                             sb.Append(typeName);
-                            sb.Append(' ');
-                            sb.Append(pi[j].Name);
+                            string? parameterName = pi[j].Name;
```

Comment:
Possible future improvement: I vaguely remember discussing this: Would it make sense to add RUC onto the `ParameterInfo.Name`? I know it can be null even without trimming, but it is an observable difference with trimming.

Reply:
ILLink will not strip parameter names from things that are visible targets of reflection. By adding RUC on all APIs that get `MethodBase` from a stack walk, we cover all scenarios where stripping parameter names could be an observable difference (it's the API that lets you get hold of a broken `MethodBase` in the first place).

Since there will be a warning on getting the `MethodBase`, we don't need another warning on `Name`.

This sets us also up to do attribute stripping correctly, similar to how we strip the parameter names: https://github.com/dotnet/linker/issues/2078 - we can remove known "useless" custom attributes from methods/fields that are not visible targets of reflection for the same reason it's safe to strip parameter names and it will not cause any warningless runtime behavior differences.